### PR TITLE
add agreed upon labels to core components

### PR DESCRIPTION
**Pull request by linki**
add `application` and `version` labels to core components. does `release` make sense?

### CODE REVIEW

File: cluster/userdata-master.yaml [hunk 1]
```diff
@@ -791,15 +812,17 @@ write_files:
       metadata:
         name: prometheus-node-exporter
         labels:
-          app: prometheus
+          application: prometheus
```
Comment (mikkeloscar) [inline cluster/userdata-master.yaml]: maybe this should be changed to `prometheus-node-exporter`

**Comment by hjacobs** [comment]
Does `k8s-app` have any special meaning which might break something (just checking..)?

**Comment by hjacobs** [comment]
:+1:

**Comment by linki** [comment]
@hjacobs I don't think the label "k8s-app" has any special meaning to kubernetes. double checked with kubernetes source code and there's nothing but test code using the label.

**Comment by linki** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**